### PR TITLE
setup design config provider

### LINKED ISSUE #3263: modify setupdesign config to get rid of cards layout on tablets

### PR DESCRIPTION
**Pull request by jiteshsingh**
- defines abstraction: NonRelationalProvider
- implements SudConfigProvider
- provides basic configs
- provides overlay config
    - non-card layout for tablets
    - navigation bar light/dark mode
    - status bar light/dark mode

fixes: https://github.com/GrapheneOS/os-issue-tracker/issues/3263

### CODE REVIEW

File: src/com/android/settings/sudconfig/SudConfigProvider.kt [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        const val METHOD_APPLY_GLIF_THEME_CONTROLLED_TRANSITION =
+            "applyGlifThemeControlledTransition"
+        const val METHOD_IS_DYNAMIC_COLOR_ENABLED = "isDynamicColorEnabled"
+        const val METHOD_IS_EMBEDDED_ACTIVITY_ONE_PANE_ENABLED = "isEmbeddedActivityOnePaneEnabled"
+        const val METHOD_IS_FULL_DYNAMIC_COLOR_ENABLED = "isFullDynamicColorEnabled"
+        const val METHOD_IS_MATERIAL_YOU_STYLE_ENABLED = "IsMaterialYouStyleEnabled"
+        const val METHOD_IS_NEUTRAL_BUTTON_STYLE_ENABLED = "isNeutralButtonStyleEnabled"
+        const val METHOD_IS_SUW_DAY_NIGHT_ENABLED = "isSuwDayNightEnabled"
+        const val METHOD_GET_DEVICE_NAME = "getDeviceName"
+        const val METHOD_GET_OVERLAY_CONFIG = "getOverlayConfig"
```
Comment (muhomorr) [inline src/com/android/settings/sudconfig/SudConfigProvider.kt]: Why declare single-use strings as separate constants? This makes the code harder to read, need to check two places to see how a method is handled. That's the reason why such strings weren't declared as separate constants in SetupDesignConfigProvider

File: src/com/android/settings/sudconfig/SudConfigProvider.kt [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        when (method) {
+            METHOD_GET_SUW_DEFAULT_THEME_STRING -> bundle.putString(method, "glif_v4_light")
+
+            METHOD_APPLY_GLIF_THEME_CONTROLLED_TRANSITION,
+            METHOD_IS_DYNAMIC_COLOR_ENABLED,
+            METHOD_IS_EMBEDDED_ACTIVITY_ONE_PANE_ENABLED,
+            METHOD_IS_FULL_DYNAMIC_COLOR_ENABLED,
+            METHOD_IS_MATERIAL_YOU_STYLE_ENABLED,
+            METHOD_IS_NEUTRAL_BUTTON_STYLE_ENABLED,
+            METHOD_IS_SUW_DAY_NIGHT_ENABLED -> bundle.putBoolean(
```
Comment (muhomorr) [inline src/com/android/settings/sudconfig/SudConfigProvider.kt]: Spreading bundle.putBoolean across 4 lines hurts readability

File: src/com/android/settings/sudconfig/SudConfigProvider.kt [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+            METHOD_GET_OVERLAY_CONFIG -> {
+                fillOverlayConfig(bundle)
+            }
+        }
+        return bundle
+    }
+
+    private fun getDeviceName(): String {
+        var name = Settings.Global.getString(
+            context!!.contentResolver,
```
Comment (muhomorr) [inline src/com/android/settings/sudconfig/SudConfigProvider.kt]: There's non-null requireContext() method, same applies to fillOverlayConfig() below

**Review by muhomorr — CHANGES_REQUESTED** [review]

File: src/com/android/settings/sudconfig/SudConfigProvider.kt (hunk 1 above)
Comment (jiteshsingh) [reply src/com/android/settings/sudconfig/SudConfigProvider.kt]: yes single use strings shouldn't be declared as constants, I don't prefer either.
but I had received comments previously by other members (on some other commits), so I thought maybe that's a standard being followed.
will remove these now, will be much better!

File: src/com/android/settings/sudconfig/SudConfigProvider.kt (hunk 2 above)
Comment (jiteshsingh) [reply src/com/android/settings/sudconfig/SudConfigProvider.kt]: done. thanks for pointing out!

File: src/com/android/settings/sudconfig/SudConfigProvider.kt (hunk 3 above)
Comment (jiteshsingh) [reply src/com/android/settings/sudconfig/SudConfigProvider.kt]: done both

File: src/com/android/settings/sudconfig/SudConfigProvider.kt (hunk 1 above)
Comment (jiteshsingh) [reply src/com/android/settings/sudconfig/SudConfigProvider.kt]: done

**Pull request merged**